src/frontend/src/handler/create_index.rs

```diff
@@ -77,77 +76,114 @@ pub(crate) fn gen_create_index_plan(
         Object::TableId(table.id.table_id),
     )])?;
 
-    let table_desc = Rc::new(table.table_desc());
-    let table_desc_map = table_desc
-        .columns
-        .iter()
-        .enumerate()
-        .map(|(x, y)| (y.name.clone(), x))
-        .collect::<HashMap<_, _>>();
-
-    let to_column_order = |(ident, order): &(Ident, OrderType)| {
-        let x = ident.real_value();
-        table_desc_map
-            .get(&x)
-            .map(|x| ColumnOrder::new(*x, *order))
-            .ok_or_else(|| ErrorCode::ItemNotFound(x).into())
-    };
-
-    let to_column_indices = |ident: &Ident| {
-        let x = ident.real_value();
-        table_desc_map
-            .get(&x)
-            .cloned()
-            .ok_or_else(|| ErrorCode::ItemNotFound(x).into())
-    };
-
-    let mut index_columns = columns
-        .iter()
-        .map(to_column_order)
-        .try_collect::<_, Vec<_>, RwError>()?;
+    let mut binder = Binder::new_for_stream(session);
+    binder.bind_table(Some(&schema_name), &table_name, None)?;
+
+    let mut index_columns_ordered_expr = vec![];
+    let mut include_columns_expr = vec![];
+    let mut distributed_columns_expr = vec![];
+    for column in columns {
+        let order_type = OrderType::from_bools(column.asc, column.nulls_first);
+        let expr_impl = binder.bind_expr(column.expr)?;
+        match &expr_impl {
+            ExprImpl::InputRef(_) => {}
+            ExprImpl::FunctionCall(func) => {
+                match func.get_expr_type() {
+                    // TODO: support more functions after verification
```

Comment:
Sorry for the unclear statement. I mean we need stable functions here, because the function result will be persistent. So Any functions unverified should be banned.

Reply:
Agree. I've recently also considered the implementation of the non-`NULL` default value of table columns, also finding that it's necessary for us to give a clear categorization of the function volatility.